evm/check-finality.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const {
+    providers: { JsonRpcProvider },
+} = ethers;
+const { Command, Option } = require('commander');
+const { printInfo, mainProcessor } = require('./utils');
+const { addBaseOptions } = require('./cli-utils');
+
+async function processCommand(_config, chain, options) {
+    const txHash = options.txHash;
+    const rpc = chain.rpc;
+    const provider = new JsonRpcProvider(rpc);
+    const tx = await provider.getTransaction(txHash);
+    const txBlock = tx.blockNumber;
+
+    while (true) {
```

Comment:
add a sleep in each iteration, to not create a huge RPC load (public RPCs might rate limit)